### PR TITLE
Update Bigtable version to 1.14.0 and switch shaded dependency to 2x

### LINKED ISSUE #2060: JanusGraph/Bigtable integration failing with current Bigtable version
Please post all questions and issues on [janusgraph-users](https://groups.google.com/forum/#!forum/janusgraph-users)
before opening a GitHub issue. Your questions will reach a wider audience there,
and if we confirm that there is a bug, then you can open a new issue.

Please include configurations and logs if available.

For confirmed bugs, please report:
- Version: 0.5.0
- Mailing list Thread URL: https://groups.google.com/forum/#!topic/janusgraph-users/I27Qk3EFJvs
- Steps to Reproduce: 

When trying to load my .properties file (`graph = JanusGraphFactory.open('conf/test.properties')`), I am also seeing the error mentioned in the mailing list thread (below). My properties file is identical to the suggested changes to work with Bigtable here: https://docs.janusgraph.org/storage-backend/bigtable/

### Stack Trace (if you have one)

```
java.lang.IllegalArgumentException: Could not instantiate implementation: org.janusgraph.diskstorage.hbase.HBaseStoreManager
	at org.janusgraph.util.system.ConfigurationUtil.instantiate(ConfigurationUtil.java:64)
	at org.janusgraph.diskstorage.Backend.getImplementationClass(Backend.java:440)
	at org.janusgraph.diskstorage.Backend.getStorageManager(Backend.java:411)
	at org.janusgraph.graphdb.configuration.builder.GraphDatabaseConfigurationBuilder.build(GraphDatabaseConfigurationBuilder.java:50)
	at org.janusgraph.core.JanusGraphFactory.open(JanusGraphFactory.java:161)
	at org.janusgraph.core.JanusGraphFactory.open(JanusGraphFactory.java:132)
	at org.janusgraph.core.JanusGraphFactory.open(JanusGraphFactory.java:79)
	at org.janusgraph.core.JanusGraphFactory$open.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
	at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:127)
	at groovysh_evaluate.run(groovysh_evaluate:3)
	at groovysh_evaluate$run.call(Unknown Source)
	at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
	at groovysh_evaluate$run.call(Unknown Source)
	at org.codehaus.groovy.tools.shell.Interpreter.evaluate(Interpreter.groovy:81)
	at org.codehaus.groovy.tools.shell.Evaluator$evaluate.call(Unknown Source)
	at org.codehaus.groovy.tools.shell.Groovysh.execute(Groovysh.groovy:201)
	
... (full stack trace in thread)

Caused by: java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
	at org.janusgraph.util.system.ConfigurationUtil.instantiate(ConfigurationUtil.java:58)
	... 64 more
Caused by: org.janusgraph.diskstorage.PermanentBackendException: Permanent failure in storage backend
	at org.janusgraph.diskstorage.hbase.HBaseStoreManager.<init>(HBaseStoreManager.java:338)
	... 69 more
Caused by: java.io.IOException: java.lang.NoSuchMethodException: com.google.cloud.bigtable.hbase1_x.BigtableConnection.<init>(org.apache.hadoop.conf.Configuration, java.util.concurrent.ExecutorService, org.apache.hadoop.hbase.security.User)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:222)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:115)
	at org.janusgraph.diskstorage.hbase.HBaseCompat1_0.createConnection(HBaseCompat1_0.java:43)
	at org.janusgraph.diskstorage.hbase.HBaseStoreManager.<init>(HBaseStoreManager.java:336)
	... 69 more
Caused by: java.lang.NoSuchMethodException: com.google.cloud.bigtable.hbase1_x.BigtableConnection.<init>(org.apache.hadoop.conf.Configuration, java.util.concurrent.ExecutorService, org.apache.hadoop.hbase.security.User)
	at java.base/java.lang.Class.getConstructor0(Class.java:3350)
	at java.base/java.lang.Class.getDeclaredConstructor(Class.java:2554)
	at org.apache.hadoop.hbase.client.ConnectionFactory.createConnection(ConnectionFactory.java:217)
	... 72 more
```

This is happening because in the last part of the stacktrace, the Bigtable Hbase 1.x client does not have a constructor mapping to the parameters listed which are called reflectively. (` java.lang.NoSuchMethodException: com.google.cloud.bigtable.hbase1_x.BigtableConnection.<init>(org.apache.hadoop.conf.Configuration, java.util.concurrent.ExecutorService, org.apache.hadoop.hbase.security.User)`)

To fix this, I'd like to propose upgrading the `bigtable-hbase-1.x-shaded` dependency to the `bigtable-hbase-2.x-shaded` version, which is compatible with the version of hbase and will not throw this error. I will create a PR for this unless there are any other ideas or objections. Thanks!

### PR DESCRIPTION
**Pull request by kolea2**
Fixes #2060 

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[doc only]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by mbrukman** [comment]
@kolea2 — could you please update the docs for 0.4.1 to specify that Bigtable storage backend doesn't work in that JanusGraph version?

I.e., update https://github.com/JanusGraph/janusgraph/blob/master/docs/changelog.md with the following changes:

* specify "N/A" for Bigtable in the `0.4.z` row, since there are no 0.4 release with Bigtable support (at least at this time—if this is fixed in the future, we can update this accordingly)
* add 1.14.0 to the list of Bigtable client versions in the `0.5.z` row
* remove any (old) Bigtable client versions from the `0.5.z` row if they're no longer supported or recommended by the Bigtable team

Please squash that change with your current commit and `git push -f` your branch.

Thanks!

File: docs/changelog.md [hunk 1]
```diff
@@ -22,8 +22,8 @@ use the latest versions of the software.
...
 | 0.2.z | 1 | 1.2.z, 2.0.z, 2.1.z, 2.2.z, 3.0.z, 3.11.z | 0.98.z, 1.0.z, 1.1.z, 1.2.z, 1.3.z | 0.9.z, 1.0.0-preZ, 1.0.0 | 1.5-1.7.z, 2.3-2.4.z, 5.y, 6.y | 5.2-5.5.z, 6.2-6.6.z, 7.y | 3.2.z | 1.6.z | 2.10.z | 
 | 0.3.z | 2 | 1.2.z, 2.0.z, 2.1.z, 2.2.z, 3.0.z, 3.11.z | 1.0.z, 1.1.z, 1.2.z, 1.3.z, 1.4.z | 1.0.0, 1.1.0, 1.1.2, 1.2.0, 1.3.0, 1.4.0 | 1.5-1.7.z, 2.3-2.4.z, 5.y, 6.y |  5.2-5.5.z, 6.2-6.6.z, 7.y | 3.3.z | 2.2.z | 2.11.z |
-| 0.4.z | 2 | 2.1.z, 2.2.z, 3.0.z, 3.11.z | 1.2.z, 1.3.z, 1.4.z, 2.1.z | 1.3.0, 1.4.0, 1.5.z, 1.6.z, 1.7.z, 1.8.z, 1.9.z, 1.10.z, 1.11.z | 5.y, 6.y | 7.y | 3.4.z | 2.2.z | 2.11.z |
-| 0.5.z | 2 | 2.1.z, 2.2.z, 3.0.z, 3.11.z | 1.2.z, 1.3.z, 1.4.z, 2.1.z | 1.3.0, 1.4.0, 1.5.z, 1.6.z, 1.7.z, 1.8.z, 1.9.z, 1.10.z, 1.11.z | 6.y, 7.y | 7.y | 3.4.z | 2.2.z | 2.11.z | 
+| 0.4.z | 2 | 2.1.z, 2.2.z, 3.0.z, 3.11.z | 1.2.z, 1.3.z, 1.4.z, 2.1.z | N/A | 5.y, 6.y | 7.y | 3.4.z | 2.2.z | 2.11.z |
+| 0.5.z | 2 | 2.1.z, 2.2.z, 3.0.z, 3.11.z | 1.2.z, 1.3.z, 1.4.z, 2.1.z | 1.3.0, 1.4.0, 1.5.z, 1.6.z, 1.7.z, 1.8.z, 1.9.z, 1.10.z, 1.11.z, 1.14.0 | 6.y, 7.y | 7.y | 3.4.z | 2.2.z | 2.11.z |
```
Comment (mbrukman) [inline docs/changelog.md]: Sorry, I should have clarified that you should add `1.14.z` (wildcard) here, not just 1.14.0 specifically.

Also, adding just 1.14.z here implies that we don't recommend 1.12.z and 1.13.z versions (e.g., they're broken or incompatible with JanusGraph) — if there's nothing wrong with them, shouldn't we add them here as well?

Comment (kolea2) [reply docs/changelog.md]: gotcha! I only added 1.14.0 as the last version in this repo was for 1.11.0 - do you recommend we test out the versions in between as well? Will update from 1.14.0 to 1.14.z.

Comment (mbrukman) [reply docs/changelog.md]: If you are able to test with 1.12 and 1.13 that would be ideal. If not, we can keep just `1.14.z`.

That said, I don't anticipate any issues with in-between client library versions, as they don't typically change semantics or behavior; it's primarily issues with dependencies or Hadoop profiles are the issues that we may run into.

Comment (kolea2) [reply docs/changelog.md]: Sounds good. For now, I've updated the `1.14.z` change for the changelog and will test out other versions in parallel to not delay this PR.

Comment (mbrukman) [reply docs/changelog.md]: That's great! We can update the docs separately, and fast-track this PR into the release.

Thanks again for your help!

**Comment by kolea2** [comment]
Thanks so much @mbrukman, @porunov !

**Comment by porunov** [comment]
`0.5.1` release artefacts have been updated to include this PR. The release is uploaded into Sonatype staging repository and the tag and GitHub release jars have been updated as well (https://github.com/JanusGraph/janusgraph/releases/tag/v0.5.1).
We are waiting for one more TSC vote to finalize the release: https://groups.google.com/forum/#!topic/janusgraph-dev/qQRc8fT5aPw

**Pull request merged**